packages/postgres/src/PostgresMessagePickupRepository.ts

```diff
@@ -0,0 +1,569 @@
+          if (this.connectionInfoCallback) {
+            const connectionInfo = await this.connectionInfoCallback(connectionId)
+
+            if (connectionInfo?.sendPushNotification) {
+              await connectionInfo.sendPushNotification(messageId)
+            }
+          } else {
+            this.logger?.error(`connectionInfoCallback is not defined`)
+            throw new Error(`connectionInfoCallback is not defined`)
+          }
+        } else if (this.dbListener) {
+          // Publish to the Pub/Sub channel if a live session exists on another instance
+          this.logger?.debug(
+            `[addMessage] Publishing new message event to Pub/Sub channel for connectionId: ${connectionId}`,
+          )
+          await this.pubSubInstance?.publish('newMessage', connectionId)
+        } else {
+          this.logger?.debug('[addMessage] No live session and no DB channel configured.')
+        }
+      }
+
+      return JSON.stringify({ messageId, receivedAt })
+    } catch (error) {
+      this.logger?.error(`[addMessage] Error during message insertion or processing: ${error}`)
+      throw new Error(`Failed to add message: ${error}`)
+    }
+  }
+
+  /**
+   * Removes specified messages from the queue for a given connection.
+   *
+   * @param {RemoveMessagesOptions} options - Options for removing messages.
+   * @param {string} options.connectionId - The ID of the connection.
+   * @param {string[]} options.messageIds - Array of message IDs to be removed.
+   * @returns {Promise<void>} A promise resolving when the operation completes.
+   */
+  public async removeMessages(options: RemoveMessagesOptions): Promise<void> {
+    const { connectionId, messageIds } = options
+    this.logger?.debug(
+      `[removeMessages] Attempting to remove messages with IDs: ${messageIds} for ConnectionId: ${connectionId}`,
+    )
+
+    // Validate messageIds
+    if (!messageIds || messageIds.length === 0) {
+      this.logger?.debug('[removeMessages] No message IDs provided. No messages will be removed.')
+      return
+    }
+
+    try {
+      // Generate placeholders for the SQL query dynamically based on messageIds length
+      const placeholders = messageIds.map((_, index) => `$${index + 2}`).join(', ')
+
+      // Construct the SQL DELETE query
+      const query = `DELETE FROM queuedmessages WHERE connectionid = $1 AND id IN (${placeholders})`
+
+      // Combine connectionId with messageIds as query parameters
+      const queryParams = [connectionId, ...messageIds]
+
+      // Execute the query
+      await this.messagesCollection?.query(query, queryParams)
+
+      this.logger?.debug(
+        `[removeMessages] Successfully removed messages with IDs: ${messageIds} for ConnectionId: ${connectionId}`,
+      )
+    } catch (error) {
+      this.logger?.error(`[removeMessages] Error occurred while removing messages: ${error}`)
+      throw new Error(`Failed to remove messages: ${error}`)
+    }
+  }
+
+  public async disposeDB() {
```

Comment:
```suggestion
  public async shutdown() {
```

This is to make it consistent with other modules in credo that use the initialize/shutdown naming

Reply:
solved!